### PR TITLE
[ Search 1280 ] SolrInformationServer refactor

### PR DESCRIPTION
**Pull request by agazzarini**
The PR contains a formal refactor of SolrInformationServer, including: 

- order and modifier (private, public) of member instances
- formatting rules (e.g. curly braces on new line, spaces) 
- private methods moved to the bottom
- unused members (methods / variables) removal
- constants (e.g. CommonParams.Q) instead of redundant literals
- unnecessary boxing / unboxing removal
- generics
- System.out removal
- a logger utility for isolating / removing conditional statements within the class
- compose methods for removing the most evident code duplications
- minimal unit test 

The PR will contain a lot of commits with a fine level of granularity, because after each change I executed the whole SearchServices test suite, in order to make sure everything was still working.

### CODE REVIEW

**Comment by tunaaksoy** [comment]
It's very difficult to review the changes in SolrInformationServer. As far as I could see there was no logic change there but just Java best practices have been applied and mostly using features from Java 7 and 8. I might have missed logic changes in there. If that's the case can you point me to those please?
Do we need the Log.java class? Can we not use slf4j?

**Comment by tunaaksoy** [comment]
Please merge the latest code changes from master and resolve the conflict in SolrInformationServer.java.

**Comment by agazzarini** [comment]
Yes, you're right. Apart formal changes, the only structural change is the unused methods removal. 
The Log class is a wrapper aroung SLF4j logger, and it avoids conditional statements in case of WARN and DEBUG messages. Instead of doing

```
if (logger.isDebugEnabled)) 
{
   logger.debug(some message)
}
```  
the Log class allows us to write only 

`LOGGER.debug(some message)`

**Comment by tunaaksoy** [comment]
> Yes, you're right. Apart formal changes, the only structural change is the unused methods removal.
> The Log class is a wrapper aroung SLF4j logger, and it avoids conditional statements in case of WARN and DEBUG messages. Instead of doing
> 
> ```
> if (logger.isDebugEnabled)) 
> {
>    logger.debug(some message)
> }
> ```
> the Log class allows us to write only
> 
> `LOGGER.debug(some message)`

Why do we need a wrapper? We use slf4j-log4j12. I would have thought that the Log4jLoggerAdapter would take care of those checks and it would be good enough with the parameterised message changes you did.

**Comment by tunaaksoy** [comment]
> Please merge the latest code changes from master and resolve the conflict in SolrInformationServer.java.

I meant merge master to your branch and not your branch to master ;) I would have liked to get Michael's or Tom's opinion as well but It's OK I think. No need to revert the changes.

File: alfresco-search/src/main/java/org/alfresco/solr/SolrInformationServer.java [hunk 1]
```diff
@@ -2555,23 +1860,18 @@ public void indexNodes(List<Node> nodes, boolean overwrite, boolean cascade) thr
                 nmdp.setNodeIds(nodeIds);
 
                 // Fetches bulk metadata
-                List<NodeMetaData> nodeMetaDatas =  repositoryClient.getNodesMetaData(nmdp, Integer.MAX_VALUE);
+                List<NodeMetaData> nodeMetaDatas = repositoryClient.getNodesMetaData(nmdp, Integer.MAX_VALUE);
 
-                NEXT_NODE: for (NodeMetaData nodeMetaData : nodeMetaDatas)
+                NEXT_NODE:
```
Comment (tpage-alfresco) [inline alfresco-search/src/main/java/org/alfresco/solr/SolrInformationServer.java]: Are we in a switch statement??  I can't see how this is used at all.

File: alfresco-search/src/main/java/org/alfresco/solr/SolrInformationServer.java [hunk 2]
```diff
@@ -4114,256 +2915,960 @@ public void reindexNodeByQuery(String query) throws IOException, AuthenticationE
...
-        private long txnFloor;
-        private Map txnCache;
+    private int portNumber(Properties properties)
+    {
+        String portNumber =
+                ofNullable(ConfigUtil.locateProperty(SOLR_PORT, properties.getProperty(SOLR_PORT)))
+                        .filter(value -> value.trim().length() > 0)
+                        .orElseGet(() -> {
+                            String jettyPort = System.getProperty("jetty.port");
+                            LOGGER.debug("Using jetty.port ", jettyPort);
```
Comment (tpage-alfresco) [inline alfresco-search/src/main/java/org/alfresco/solr/SolrInformationServer.java]: Missing braces in string?

File: alfresco-search/src/main/java/org/alfresco/solr/SolrInformationServer.java [hunk 3]
```diff
@@ -4114,256 +2915,960 @@ public void reindexNodeByQuery(String query) throws IOException, AuthenticationE
...
+            {
+                Thread.sleep(100);
+            }
+            catch (InterruptedException e)
+            {
+                // I don't think we are concerned with this exception.
+            }
 
-            if(txnId >= txnFloor)
+            if(System.currentTimeMillis() - startTime > 120000)
```
Comment (tpage-alfresco) [inline alfresco-search/src/main/java/org/alfresco/solr/SolrInformationServer.java]: There's a constant for two minutes we could use at the top of this file.

File: alfresco-search/src/main/java/org/alfresco/solr/SolrInformationServer.java [hunk 4]
```diff
@@ -4114,256 +2915,960 @@ public void reindexNodeByQuery(String query) throws IOException, AuthenticationE
...
+        switch (AuthorityType.getAuthorityType(authority))
         {
-            currentLongs = context.reader().getNumericDocValues(FIELD_INTXID);
+            case GROUP:
+            case EVERYONE:
+            case GUEST:
+                if (tenant.length() != 0)
+                {
+                    return authority + "@" + tenant;
+                }
```
Comment (tpage-alfresco) [inline alfresco-search/src/main/java/org/alfresco/solr/SolrInformationServer.java]: We should probably have a break statement here.  Would this switch statement be simpler as an if statement?

File: alfresco-search/src/main/java/org/alfresco/solr/SolrInformationServer.java [hunk 5]
```diff
@@ -4114,256 +2915,960 @@ public void reindexNodeByQuery(String query) throws IOException, AuthenticationE
...
+
+                if (dec.isDefinitionExists(qName))
+                {
+                    dataForSkippingDescendantDocs.add(qName);
+                }
+            }
         }
 
-        public void collect(int doc) throws IOException
+        if (null == skippingDocsQueryString)
```
Comment (tpage-alfresco) [inline alfresco-search/src/main/java/org/alfresco/solr/SolrInformationServer.java]: Is skippingDocsQueryString a member variable?  Using a member variable in a private method in such a large class seems fairly unreadable.  Perhaps this method could return the string instead?

File: alfresco-search/src/main/java/org/alfresco/solr/SolrInformationServer.java [hunk 6]
```diff
@@ -4114,256 +2915,960 @@ public void reindexNodeByQuery(String query) throws IOException, AuthenticationE
...
-            return size() > this.maxSize;
+    private String getFieldValueString(SolrDocument doc, String fieldName)
+    {
+        IndexableField field = (IndexableField)doc.getFieldValue(fieldName);
+        String value = null;
+        if (field != null)
+        {
+            value = field.stringValue();
         }
+        return value;
```
Comment (tpage-alfresco) [inline alfresco-search/src/main/java/org/alfresco/solr/SolrInformationServer.java]: We could use a ternary operator or a helper method that provides defaulting here?
```
return (field != null ? field.stringValue() : null);
```

**Pull request merged**